pallets/parachain-staking/src/lib.rs

```diff
@@ -465,6 +493,9 @@ pub mod pallet {
 			weight
 		}
 		fn on_finalize(_n: T::BlockNumber) {
+			let author = T::BlockAuthor::get();
```

Comment:
It wopuld be better to do that on an inherent or at the beginning of the next block.
Here it's too expensive to add 2 reads and 1 write in on_finalize.
Also note that weights cosumes by on_finalize should be accounted in on_initialize